### DISCUSSION TITLE
Please provide a complete guideline for contributing new devices

### DISCUSSION THREAD

**Discussion by aheadlead** [rytilahti/python-miio#1056] [Q&A] [answered]
Hi @rytilahti ,

Thanks your all efforts on the phenomenal library python-miio, it saved tons of my time on DIY.

I bought an unsupported smart plug (Gosund socket, cuco.plug.cp2) with MIoT-Spec V2 protocol support and write some codes to support this device now on my local repo. I suggest you provide a guideline (or check list) to save both you and contributors time. Once I have fully tested, I will create a pull request for you.

Thanks,
Yulan

**Comment by rytilahti** [comment, marked as answer]
Hi @aheadlead and thanks for reminding me about this. I added some information about contributing to https://github.com/rytilahti/python-miio/pull/1139 . It's not anyway complete, but I hope it'll get extended in the future :-) So if you have some specific questions, feel free to ask and I'll add more information there.
